analyzers/ibananalyzer.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+import re
+
+from .regexanalyzer import RegexAnalyzer
+
+
+class IBANAnalyzer(RegexAnalyzer):
+    def __init__(self, action, validate=False):
+        # Regex adapted from https://stackoverflow.com/a/44657292/3621482
+        regex = r"([A-Z]{2}[ \-]?[0-9]{2})(?!=(?:[ \-]?[A-Z0-9]){9,30}$)" \
+                r"((?:[ \-]?[A-Z0-9]{3,5}){2,7})([ \-]?[A-Z0-9]{1,3})?"
+        super().__init__(action, regex)
+        self.validate = validate
+
+    def match(self, paste):
+        paste_content = paste.body
+        search_result = self.regex.search(paste_content)
+
+        if search_result is None:
+            return False
+
+        if self.validate:
+            matched_string = search_result.group()
+            return self._validate_iban(matched_string)
+
+        return True
+
+    def _validate_iban(self, potential_iban):
+        """Checks if the given string could be a valid IBAN. Adapted from https://rosettacode.org/wiki/IBAN#Python."""
+
+        # Ensure upper alphanumeric input
+        potential_iban = potential_iban.replace(' ', '').replace('\t', '').replace('-', '')
+        if not re.match(r"^[\dA-Z]+$", potential_iban):
+            return False
+
+        # Validate if length matches the expected length based on the country code
+        country_code = potential_iban[:2]
+        if len(potential_iban) != self._iban_length_by_country.get(country_code, None):
+            return False
+
+        # Official validation ( https://en.wikipedia.org/wiki/International_Bank_Account_Number#Validating_the_IBAN )
+        rearranged_iban = potential_iban[4:] + potential_iban[:4]
+        integer_iban = int(''.join(str(int(ch, 36)) for ch in rearranged_iban))  # BASE 36: 0..9,A..Z -> 0..35
+        return integer_iban % 97 == 1
+
+    _iban_length_by_country = dict(
```

Comment:
Wow, that's a lot of countries / different lenghts. I thought they are a bit more ... unified.